compose-builder/add-app-rfid-llrp-inventory.yml

```diff
@@ -30,6 +30,7 @@ services:
       - consul
       - data
     read_only: true
+    restart: always
```

Comment:
I would put max number of times for restart; not "always" which may cause system memory exhaustion if restarting too many times.

Reply:
That isn't a choice with `restart` and the other policies don't do what we need.
https://github.com/edgexfoundry/app-rfid-llrp-inventory/pull/52
That said it looks like with version 3.9 there may be something with `restart_policy` but not sure if that is just for Swarm. Will investigate.
https://docs.docker.com/compose/compose-file/compose-file-v3/#restart_policy

Reply:
@jim-wang-intel , not that doesn't work
```
maximum retry count cannot be used with restart policy 'always'
```
`on-failure` doesn't work because the services exit gracefully on reboot due to our handling of SIGTERM.

Reply:
well, that is kind of interesting, it should do for  `on-failure` and if our services exited gracefully then it considers as OK condition right?  Otherwise, maybe we should fix that as it seems to suggest design flaws...

Reply:
@jim-wang-intel, Not just our services. Kong, Consul, etc. work the same way so our only choice if we want all the services to restart after a reboot is to use `always`

Reply:
ok, I guess the original main concern is memory exhaustion from restarting too many time. if reboot is our main use case, probably it is fine.  If memory becomes an issue, we might also use "resource limit" on docker-compose to restrict the memory exhaustion?   I guess we could experiment with that infinite restarting case when something is broken to see how memory behaves. 
but for now, I am ok with it.